# Factores que determinan el ingreso laboral en Colombia
## Notebook 01 — Carga, exploración y limpieza

**Fuente de datos:** [GEIH 2025 — DANE](https://microdatos.dane.gov.co/index.php/catalog/853)  
**Archivos utilizados:** Características generales, seguridad social en salud y educación + Ocupados (Enero — Junio 2025)  
**Autora:** Katherin Liceth Reyes Enciso

## Objetivo

Identificar qué factores influyen en el ingreso laboral de los colombianos usando datos de la Gran Encuesta Integrada de Hogares (GEIH) del DANE, correspondientes al primer semestre de 2025 (enero a junio).

Los datos provienen de dos tablas de la GEIH:
- **Ocupados:** información laboral de las personas que trabajan (ingreso, sector, contrato, horas, etc.)
- **Características generales:** información demográfica (sexo, edad, nivel educativo, departamento, etc.)

Ambas tablas se cruzan usando las llaves `DIRECTORIO`, `SECUENCIA_P`, `ORDEN`, `HOGAR` y `MES`, que en conjunto identifican a cada persona encuestada en un mes específico.

## Variables utilizadas

### Variables de identificación y contexto

| Variable | Descripción | Tabla de origen |
|----------|-------------|-----------------|
| DIRECTORIO | Identificador de la vivienda | Ambas |
| SECUENCIA_P | Identificador del hogar dentro del operativo de la encuesta | Ambas |
| ORDEN | Número de la persona dentro del hogar (1, 2, 3...) | Ambas |
| HOGAR | Número del hogar dentro de la vivienda (puede haber más de uno) | Ambas |
| MES | Mes de la encuesta (1 a 6) | Ambas |
| DPTO | Código del departamento | Ocupados |
| CLASE | Zona: 1 = Urbana, 2 = Rural | Ocupados |
| FEX_C18 | Factor de expansión. Indica cuántas personas de la población representa cada encuestado | Ocupados |

### Variable objetivo

| Variable | Descripción | Valores |
|----------|-------------|---------|
| INGLABO | Ingreso laboral total mensual  | Valor en pesos colombianos |

### Variables demográficas (de Características generales)

| Variable | Descripción | Valores |
|----------|-------------|---------|
| P3271 | Sexo al nacer | 1 = Hombre,  2 = Mujer |
| P6040 | Edad en años cumplidos | 15 a 100 |
| P3042 | Mayor nivel educativo alcanzado | 1 = Ninguno,<br> 2 = Preescolar<br> 3 = Básica primaria (1°-5°)  <br> 4 = Básica secundaria (6°-9°) <br>5 = Media académica <br> 6 = Media técnica <br>7 = Normalista <br>8 = Técnica profesional <br> 9 = Tecnológica <br>10 = Universitaria,<br> 11 = Especialización <br> 12 = Maestría<br> 13 = Doctorado <br>99 = No sabe/no informa |

### Variables laborales (de Ocupados)

| Variable | Descripción | Valores |
|----------|-------------|---------|
| P6800 | Horas trabajadas por semana | Número |
| RAMA2D_R4 | Sector económico (código CIIU a 2 dígitos) | Código numérico |
| P6430 | Posición ocupacional | 1 = Empleado particular <br>2 = Empleado gobierno <br>3 = Empleado doméstico<br> 4 = Cuenta propia<br> 5 = Patrón o empleador<br> 6 = Familiar sin pago<br> 7 = Jornalero o peón<br> 8 = Otro |
| P6440 | ¿Tiene algún tipo de contrato? | 1 = Sí, 2 = No |
| P6450 | Tipo de contrato | 1 = Verbal<br> 2 = Escrito<br> 9 = No sabe, no informa |
| P6460 | Duración del contrato (solo si es escrito) | 1 = Indefinido<br> 2 = Fijo<br> 9 = No sabe, no informa |
| P6920 | ¿Cotiza pensión? | 1 = Sí, 2 = No |
| P6990 | ¿Tiene ARL? (Aseguradora de Riesgos Laborales) | 1 = Sí, 2 = No |
| P6426 | Antigüedad en la empresa (meses) | Número |
| P3069 | Tamaño de la empresa | 1: Trabaja solo <br>2: 2 a 3 personas<br> 3: 4 a 5 personas<br> 4: 6 a 10 personas<br> 5: 11 a 19 personas<br> 6: 20 a 30 personas<br> 7: 31 a 50 personas<br> 8: 51 a 100 personas<br> 9: 101 a 200 personas<br> 10: 201 o más personas |

### Clasificación del sector económico (RAMA2D_R4)

Los códigos de esta variable corresponden a la Clasificación Industrial Internacional Uniforme (CIIU Rev. 4 A.C. 2022) del DANE. La tabla muestra cómo se agrupan:

| Códigos | Descripción |
|---------|-------------|
| 01 – 03 | Agricultura, ganadería, caza, silvicultura y pesca |
| 05 – 09 | Explotación de minas y canteras |
| 10 – 33 | Industrias manufactureras |
| 35 | Suministro de electricidad, gas, vapor y aire acondicionado |
| 36 – 39 | Distribución de agua; evacuación y tratamiento de aguas residuales, gestión de desechos y actividades de saneamiento ambiental |
| 41 – 43 | Construcción |
| 45 – 47 | Comercio al por mayor y al por menor; reparación de vehículos automotores y motocicletas |
| 49 – 53 | Transporte y almacenamiento |
| 55 – 56 | Alojamiento y servicios de comida |
| 58 – 63 | Información y comunicaciones |
| 64 – 66 | Actividades financieras y de seguros |
| 68 | Actividades inmobiliarias |
| 69 – 75 | Actividades profesionales, científicas y técnicas |
| 77 – 82 | Actividades de servicios administrativos y de apoyo |
| 84 | Administración pública y defensa; planes de seguridad social de afiliación obligatoria |
| 85 | Educación |
| 86 – 88 | Actividades de atención de la salud humana y de asistencia social |
| 90 – 93 | Actividades artísticas, de entretenimiento y recreación |
| 94 – 96 | Otras actividades de servicios |
| 97 – 98 | Actividades de los hogares en calidad de empleadores |
| 99 | Actividades de organizaciones y entidades extraterritoriales |

Fuente: [CIIU Rev. 4 A.C. 2022 — DANE](https://www.dane.gov.co/files/sen/nomenclatura/ciiu/CIIU_Rev_4_AC2022.pdf)

## Preguntas de investigación

Estas son las preguntas que buscamos responder con el análisis:

| # | Pregunta | Variables involucradas |
|---|----------|----------------------|
| 1 | ¿Cómo se distribuye el ingreso laboral en Colombia? | INGLABO |
| 2 | ¿El nivel educativo se refleja en el ingreso? | P3042 (nivel educativo), INGLABO |
| 3 | ¿Existe diferencia salarial entre hombres y mujeres? | P3271 (sexo), INGLABO |
| 4 | ¿Se mantiene esa diferencia al controlar por nivel educativo? | P3271 (sexo), P3042 (nivel educativo), INGLABO |
| 5 | ¿Qué sectores económicos pagan más? | RAMA2D_R4 (sector), INGLABO |
| 6 | ¿La zona (urbana/rural) afecta el ingreso? | CLASE (zona), INGLABO |
| 7 | ¿Tener contrato escrito implica ganar más? | P6440, P6450, P6460 (contrato), INGLABO |
| 8 | ¿Qué departamentos tienen mayores ingresos? | DPTO (departamento), INGLABO |
| 9 | ¿La posición ocupacional influye en el ingreso? | P6430 (posición), INGLABO |
| 10 | ¿Las horas trabajadas correlacionan con el ingreso? | P6800 (horas/semana), INGLABO |
| 11 | ¿La antigüedad en la empresa influye? | P6426 (antigüedad), INGLABO |

Las preguntas se abordan en el análisis exploratorio con Python (Notebook 02) y algunas se podrán visualizar de forma interactiva en el dashboard de Power BI.

In [ ]:
import os
import pandas as pd
import numpy as np
import warnings
warnings.filterwarnings('ignore')

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


### Cargar los datos en formato csv

Los datos se encuentran de forma independiente, por lo que cada mes tendrá su archivo correspondiente de características y ocupados, esto implica que se tendran 6 CSV por categoría.

In [ ]:
BASE = '/content/drive/MyDrive/Factores que determinan el ingreso laboral en Colombia'

meses = ['Enero 2025', 'Febrero 2025', 'Marzo 2025',
         'Abril 2025', 'Mayo 2025', 'Junio 2025']

ocupados = {}

for mes in meses:
    ruta = f'{BASE}/{mes}/CSV/Ocupados.CSV'
    ocupados[mes] = pd.read_csv(ruta, sep=';', encoding='latin-1', low_memory=False)
    print(f'✅ Ocupados {mes}: {len(ocupados[mes]):,} filas, {ocupados[mes].shape[1]} columnas')

print(f'\n📊 Total de meses cargados: {len(ocupados)}')

✅ Ocupados Enero 2025: 28,317 filas, 202 columnas
✅ Ocupados Febrero 2025: 29,211 filas, 202 columnas
✅ Ocupados Marzo 2025: 29,471 filas, 202 columnas
✅ Ocupados Abril 2025: 29,514 filas, 202 columnas
✅ Ocupados Mayo 2025: 29,957 filas, 202 columnas
✅ Ocupados Junio 2025: 29,706 filas, 202 columnas

📊 Total de meses cargados: 6


In [ ]:
ocupados['Febrero 2025'].head()

,PERIODO,MES,PER,DIRECTORIO,SECUENCIA_P,ORDEN,HOGAR,REGIS,AREA,CLASE,...,P1881,P1881S1,P1882,P7240,P7240S1,OCI,INGLABO,RAMA2D_R4,RAMA4D_R4,OFICIO_C8
0,20250209,2,2025,8115691,1,2,1,60,NaN,2,...,11,NaN,60.0,2,NaN,1,1600000.0,94,9491,3114
1,20250209,2,2025,8115693,1,1,1,60,NaN,2,...,12,NaN,15.0,5,NaN,1,1500000.0,97,9700,9111
2,20250209,2,2025,8115695,1,1,1,60,NaN,2,...,8,NaN,20.0,2,NaN,1,3200000.0,43,4321,3113
3,20250209,2,2025,8115695,1,2,1,60,NaN,2,...,2,NaN,60.0,2,NaN,1,1423500.0,85,8530,1412
4,20250209,2,2025,8115695,1,3,1,60,NaN,2,...,2,NaN,20.0,2,NaN,1,1423500.0,47,4759,4311


In [ ]:
CSV_CARACT = 'Características generales, seguridad social en salud y educación.CSV'

caracteristicas = {}

for mes in meses:
    ruta = f'{BASE}/{mes}/CSV/{CSV_CARACT}'
    caracteristicas[mes] = pd.read_csv(ruta, sep=';', encoding='latin-1', low_memory=False)
    print(f'✅ Características {mes}: {len(caracteristicas[mes]):,} filas, {caracteristicas[mes].shape[1]} columnas')

print(f'\n📊 Total de meses cargados: {len(caracteristicas)}')

✅ Características Enero 2025: 67,857 filas, 55 columnas
✅ Características Febrero 2025: 67,953 filas, 55 columnas
✅ Características Marzo 2025: 68,083 filas, 55 columnas
✅ Características Abril 2025: 68,253 filas, 55 columnas
✅ Características Mayo 2025: 68,421 filas, 55 columnas
✅ Características Junio 2025: 67,856 filas, 55 columnas

📊 Total de meses cargados: 6


In [ ]:
caracteristicas['Febrero 2025'].head()

,PERIODO,MES,PER,DIRECTORIO,SECUENCIA_P,ORDEN,HOGAR,REGIS,AREA,CLASE,...,P6170,P3041,P3042,P3042S1,P3042S2,P3043,P3043S1,P3038,P3039,POB_MAY18
0,20250209,2,2025,8115690,1,1,1,10,NaN,2,...,2.0,NaN,3.0,3.0,NaN,NaN,NaN,2.0,1.0,1.0
1,20250209,2,2025,8115690,1,2,1,10,NaN,2,...,2.0,NaN,3.0,4.0,NaN,NaN,NaN,1.0,2.0,1.0
2,20250209,2,2025,8115691,1,1,1,10,NaN,2,...,2.0,NaN,8.0,4.0,119.0,5.0,119.0,1.0,2.0,1.0
3,20250209,2,2025,8115691,1,2,1,10,NaN,2,...,2.0,NaN,8.0,4.0,714.0,5.0,714.0,2.0,1.0,1.0
4,20250209,2,2025,8115691,1,3,1,10,NaN,2,...,1.0,1.0,4.0,1.0,NaN,NaN,NaN,NaN,NaN,NaN


### Verificación de columnas y unión de los archivos CSV

Se verifica que los archivos de cada categoría tengan las mismas columnas y en el mismo orden para luego concatenar verticalmente con el obejtivo de obtener solo una tabla por categoría.

In [ ]:
# Verificar Ocupados
columnas_ref_oc = list(ocupados['Enero 2025'].columns)
print('=== OCUPADOS ===')
for mes, df in ocupados.items():
    mismas = list(df.columns) == columnas_ref_oc
    print(f'  {mes}: {df.shape[1]} columnas — {"✅ Mismas y mismo orden" if mismas else "❌ Diferente"}')
    if not mismas:
        # Revisar si son las mismas pero en otro orden
        if set(df.columns) == set(columnas_ref_oc):
            print(f'    ⚠️ Mismas columnas pero en orden diferente')
        else:
            faltan = set(columnas_ref_oc) - set(df.columns)
            sobran = set(df.columns) - set(columnas_ref_oc)
            if faltan: print(f'    Faltan: {faltan}')
            if sobran: print(f'    Sobran: {sobran}')

# Verificar Características
columnas_ref_cg = list(caracteristicas['Enero 2025'].columns)
print('\n=== CARACTERÍSTICAS ===')
for mes, df in caracteristicas.items():
    mismas = list(df.columns) == columnas_ref_cg
    print(f'  {mes}: {df.shape[1]} columnas — {"✅ Mismas y mismo orden" if mismas else "❌ Diferente"}')
    if not mismas:
        if set(df.columns) == set(columnas_ref_cg):
            print(f'    ⚠️ Mismas columnas pero en orden diferente')
        else:
            faltan = set(columnas_ref_cg) - set(df.columns)
            sobran = set(df.columns) - set(columnas_ref_cg)
            if faltan: print(f'    Faltan: {faltan}')
            if sobran: print(f'    Sobran: {sobran}')

=== OCUPADOS ===
  Enero 2025: 202 columnas — ✅ Mismas y mismo orden
  Febrero 2025: 202 columnas — ✅ Mismas y mismo orden
  Marzo 2025: 202 columnas — ✅ Mismas y mismo orden
  Abril 2025: 202 columnas — ✅ Mismas y mismo orden
  Mayo 2025: 202 columnas — ✅ Mismas y mismo orden
  Junio 2025: 202 columnas — ✅ Mismas y mismo orden

=== CARACTERÍSTICAS ===
  Enero 2025: 55 columnas — ✅ Mismas y mismo orden
  Febrero 2025: 55 columnas — ✅ Mismas y mismo orden
  Marzo 2025: 55 columnas — ✅ Mismas y mismo orden
  Abril 2025: 55 columnas — ✅ Mismas y mismo orden
  Mayo 2025: 55 columnas — ✅ Mismas y mismo orden
  Junio 2025: 55 columnas — ✅ Mismas y mismo orden


In [ ]:
orden = ['Enero 2025', 'Febrero 2025', 'Marzo 2025',
         'Abril 2025', 'Mayo 2025', 'Junio 2025']

ocupados_total = pd.concat([ocupados[mes] for mes in orden], ignore_index=True)
caracteristicas_total = pd.concat([caracteristicas[mes] for mes in orden], ignore_index=True)

print(f'Ocupados total:          {len(ocupados_total):,} filas x {ocupados_total.shape[1]} columnas')
print(f'Características total:   {len(caracteristicas_total):,} filas x {caracteristicas_total.shape[1]} columnas')

#print(f'\nRegistros por mes (Ocupados):')
#print(df_ocupados_total.groupby('MES', sort=False).size())

Ocupados total:          176,176 filas x 202 columnas
Características total:   408,423 filas x 55 columnas


### Selección de columnas y cruce de tablas

Dado que en conjunto, ambas tablas tienen un total de 257 columnas, se seleccionaran solo las columnas necesarias de cada categoría y posteriormente se uniran las dos tablas por medio de sus llaves primarias formando una tabla unificada con la información necesaria para el análisis

In [ ]:
# Columnas que necesitamos de Ocupados
cols_ocupados = [
    'DIRECTORIO', 'SECUENCIA_P', 'ORDEN', 'HOGAR', 'MES',  # llaves
    'CLASE', 'DPTO', 'FEX_C18',                              # ubicación y factor de expansión
    'INGLABO', 'P6800',                                      # ingreso y horas
    'RAMA2D_R4',                                               # sector económico
    'P6430',                                                   # posición ocupacional
    'P6440', 'P6450', 'P6460',                                # contrato
    'P6920', 'P6990',                                          # pensión y ARL
    'P6426', 'P3069'                                           # antigüedad y tamaño empresa
]

# Columnas que necesitamos de Características
cols_caract = [
    'DIRECTORIO', 'SECUENCIA_P', 'ORDEN', 'HOGAR', 'MES',  # llaves
    'P3271',                                                  # sexo
    'P6040',                                                  # edad
    'P3042'                                                  # nivel educativo
]

ocupados_sel = ocupados_total[cols_ocupados].copy()
caract_sel = caracteristicas_total[cols_caract].copy()

print(f'Ocupados seleccionado:          {ocupados_sel.shape[0]:,} filas x {ocupados_sel.shape[1]} columnas')
print(f'Características seleccionado:   {caract_sel.shape[0]:,} filas x {caract_sel.shape[1]} columnas')

Ocupados seleccionado:          176,176 filas x 19 columnas
Características seleccionado:   408,423 filas x 8 columnas


In [ ]:
ocupados_sel.head()

,DIRECTORIO,SECUENCIA_P,ORDEN,HOGAR,MES,CLASE,DPTO,FEX_C18,INGLABO,P6800,RAMA2D_R4,P6430,P6440,P6450,P6460,P6920,P6990,P6426,P3069
0,8086872,1,1,1,1,2,13,2184.752283,1040000.0,24,97,3,1,1.0,NaN,2,2,10,1
1,8086874,1,2,1,1,2,13,1208.490211,700000.0,60,1,4,2,NaN,NaN,2,2,480,1
2,8086874,1,3,1,1,2,13,1208.490211,1450000.0,50,1,1,1,1.0,NaN,2,2,12,2
3,8086876,1,2,1,1,2,13,2420.064922,350000.0,40,47,4,2,NaN,NaN,2,2,216,1
4,8086876,1,3,1,1,2,13,2420.064922,1350000.0,50,1,7,1,1.0,NaN,2,2,144,2


In [ ]:
caract_sel.head()

,DIRECTORIO,SECUENCIA_P,ORDEN,HOGAR,MES,P3271,P6040,P3042
0,8086872,1,1,1,1,2,54,3.0
1,8086874,1,1,1,1,2,79,3.0
2,8086874,1,2,1,1,1,60,4.0
3,8086874,1,3,1,1,1,59,4.0
4,8086875,1,1,1,1,2,80,10.0


In [ ]:
llaves = ['DIRECTORIO', 'SECUENCIA_P', 'ORDEN', 'HOGAR', 'MES']

datos = ocupados_sel.merge(caract_sel, on=llaves, how='left')

print(f'Ocupados antes del cruce:   {len(ocupados_sel):,} filas')
print(f'Después del cruce:          {len(datos):,} filas')
print(f'Columnas:                   {datos.shape[1]}')

# Verificar que el cruce funcionó bien
sin_match = datos['P6040'].isna().sum()
pct_match = (1 - sin_match / len(datos)) * 100
print(f'\nPersonas con datos demográficos: {pct_match:.1f}%')
print(f'Personas sin match: {sin_match:,}')

Ocupados antes del cruce:   176,176 filas
Después del cruce:          176,176 filas
Columnas:                   22

Personas con datos demográficos: 100.0%
Personas sin match: 0


In [ ]:
datos.head()

,DIRECTORIO,SECUENCIA_P,ORDEN,HOGAR,MES,CLASE,DPTO,FEX_C18,INGLABO,P6800,...,P6440,P6450,P6460,P6920,P6990,P6426,P3069,P3271,P6040,P3042
0,8086872,1,1,1,1,2,13,2184.752283,1040000.0,24,...,1,1.0,NaN,2,2,10,1,2,54,3.0
1,8086874,1,2,1,1,2,13,1208.490211,700000.0,60,...,2,NaN,NaN,2,2,480,1,1,60,4.0
2,8086874,1,3,1,1,2,13,1208.490211,1450000.0,50,...,1,1.0,NaN,2,2,12,2,1,59,4.0
3,8086876,1,2,1,1,2,13,2420.064922,350000.0,40,...,2,NaN,NaN,2,2,216,1,1,45,4.0
4,8086876,1,3,1,1,2,13,2420.064922,1350000.0,50,...,1,1.0,NaN,2,2,144,2,1,44,5.0


### Exploración de los datos

Antes de limpiar los datos necesitamos entender qué tenemos. En esta sección revisamos tres cosas:

1. **Valores nulos:** ¿hay columnas con datos faltantes? ¿cuántos? Esto nos dice qué columnas debemos tratar con cuidado.
2. **Ingreso laboral (INGLABO):** es la variable que queremos explicar. Necesitamos saber cómo se distribuye, si hay ceros o valores extremos que puedan distorsionar el análisis.
3. **Variables demográficas y laborales:** sexo, edad, educación, zona, posición ocupacional. Revisamos qué valores tiene cada una y cuántas personas hay en cada categoría.

In [ ]:
# Revisamos si hay columnas con valores faltantes (nulos)
# Esto es importante porque un nulo en ingreso o edad afecta el análisis
print(f'Filas: {len(datos):,}')
print(f'Columnas: {datos.shape[1]}')
print(f'\n=== VALORES NULOS ===')
nulos = datos.isnull().sum()
nulos = nulos[nulos > 0].sort_values(ascending=False)
if len(nulos) > 0:
    for col, n in nulos.items():
        print(f'  {col}: {n:,} nulos ({n/len(datos)*100:.1f}%)')
else:
    print('  No hay valores nulos')

Filas: 176,176
Columnas: 22

=== VALORES NULOS ===
  P6460: 107,223 nulos (60.9%)
  P6450: 70,678 nulos (40.1%)
  INGLABO: 4,329 nulos (2.5%)


**¿Qué significan estos nulos?**

- **P6460** (duración del contrato): 60.9% de nulos. Esta pregunta es condicional, solo se activa cuando la persona tiene contrato escrito (P6450 = 2). Si no tiene contrato o lo tiene verbal, la pregunta no se le hace. Por eso aparece como nulo.
- **P6450** (tipo de contrato): 40.1% de nulos. Mismo caso: solo se pregunta a quienes respondieron que sí tienen contrato (P6440 = 1).
- **INGLABO** (ingreso laboral): 2.5% de nulos. Son personas ocupadas que no reportaron ingreso. Según la documentación del DANE, los códigos 98 y 99 en variables de ingreso se convierten en nulos para evitar que se sumen con los valores numéricos.

En resumen: la mayoría de los nulos no son datos perdidos sino preguntas condicionales que no aplican a ciertas personas. Esto es normal en encuestas con flujos de preguntas.

In [ ]:
print('=== INGRESO LABORAL (INGLABO) ===')
print(datos['INGLABO'].describe())
print(f'\nValores nulos:     {datos["INGLABO"].isna().sum():,}')
print(f'Valores = 0:       {(datos["INGLABO"] == 0).sum():,}')
print(f'Valores negativos: {(datos["INGLABO"] < 0).sum():,}')
print(f'Valores > 0:       {(datos["INGLABO"] > 0).sum():,}')

=== INGRESO LABORAL (INGLABO) ===
count    1.718470e+05
mean     1.725962e+06
std      2.230092e+06
min      0.000000e+00
25%      8.000000e+05
50%      1.423000e+06
75%      1.800000e+06
max      1.030000e+08
Name: INGLABO, dtype: float64

Valores nulos:     4,329
Valores = 0:       1,378
Valores negativos: 0
Valores > 0:       170,469


**¿Qué nos dice el ingreso?**

- Hay **170,469 personas** con ingreso mayor a cero, **4,329 sin dato** y **1,378 con ingreso cero**.
- La **mediana** (el valor del medio) es **1,423,000** y el **promedio** es **1,725,962**. Que el promedio sea mayor que la mediana indica que hay pocas personas con ingresos muy altos que jalan el promedio hacia arriba. Esto es típico en distribuciones de ingreso.
- El **máximo** es **$103,000,000**, un valor extremo frente a la mediana.
- Para la limpieza eliminaremos los registros con ingreso nulo o cero, ya que no podemos analizar factores del ingreso si no hay ingreso reportado.

In [ ]:
print('=== SEXO (P3271) ===')
print(datos['P3271'].value_counts())
print('(1=Hombre, 2=Mujer)')

print('\n=== EDAD (P6040) ===')
print(datos['P6040'].describe())

print('\n=== NIVEL EDUCATIVO (P3042) ===')
print(datos['P3042'].value_counts().sort_index())


print('\n=== ZONA (CLASE) ===')
print(datos['CLASE'].value_counts())

print('\n=== POSICIÓN OCUPACIONAL (P6430) ===')
print(datos['P6430'].value_counts().sort_index())

=== SEXO (P3271) ===
P3271
1    98242
2    77934
Name: count, dtype: int64
(1=Hombre, 2=Mujer)

=== EDAD (P6040) ===
count    176176.000000
mean         41.887085
std          14.140972
min          15.000000
25%          30.000000
50%          41.000000
75%          52.000000
max         100.000000
Name: P6040, dtype: float64

=== NIVEL EDUCATIVO (P3042) ===
P3042
1.0      4501
2.0        13
3.0     30396
4.0     18686
5.0     56096
6.0      4347
7.0       266
8.0     16315
9.0      7640
10.0    28383
11.0     6101
12.0     2991
13.0      440
99.0        1
Name: count, dtype: int64

=== ZONA (CLASE) ===
CLASE
1    154693
2     21483
Name: count, dtype: int64

=== POSICIÓN OCUPACIONAL (P6430) ===
P6430
1    74002
2     9134
3     5957
4    76089
5     4005
6     2895
7     4025
8       69
Name: count, dtype: int64


**¿Qué nos dicen las variables demográficas y laborales?**

- **Sexo (P3271):** 98,242 hombres (55.8%) y 77,934 mujeres (44.2%). Hay más hombres ocupados que mujeres.
- **Edad (P6040):** va de 15 a 100 años, con una mediana de 41.
- **Nivel educativo (P3042):** las categorías con más personas son Media académica (código 5 con 56,096), Básica primaria (código 3 con 30,396) y Universitaria (código 10 con 28,383). El código 99 (no sabe/no informa) solo tiene 1 persona.
- **Zona (CLASE):** 154,693 personas en cabecera (código 1) y 21,483 en resto (código 2).
- **Posición ocupacional (P6430):** las dos categorías más grandes son Cuenta propia (76,089) y Empleado particular (74,002), que juntas representan el 85% de los ocupados.

### Exploración y limpieza de datos

En esta sección se esploraran los datos seleccionamos y se filtraran los registros que no se pueden usar en el análisis.

In [ ]:
print(f'Registros iniciales: {len(datos):,}')

# 1. Quitar personas sin ingreso reportado o con ingreso = 0
datos_limpio = datos[(datos['INGLABO'].notna()) & (datos['INGLABO'] > 0)].copy()
print(f'Sin nulos ni ceros en INGLABO: {len(datos_limpio):,}  (eliminados: {len(datos) - len(datos_limpio):,})')

# 2. Filtrar mayores de 18 años
antes = len(datos_limpio)
datos_limpio = datos_limpio[datos_limpio['P6040'] >= 18]
print(f'Solo mayores de 18: {len(datos_limpio):,}  (eliminados: {antes - len(datos_limpio):,})')

# 3. Quitar el código 99 en nivel educativo (No sabe / no informa)
antes = len(datos_limpio)
datos_limpio = datos_limpio[datos_limpio['P3042'] != 99]
print(f'Sin educación "No sabe": {len(datos_limpio):,}  (eliminados: {antes - len(datos_limpio):,})')

print(f'\n✅ Total eliminados: {len(datos) - len(datos_limpio):,}')
print(f'✅ Registros finales: {len(datos_limpio):,}')

Registros iniciales: 176,176
Sin nulos ni ceros en INGLABO: 170,469  (eliminados: 5,707)
Solo mayores de 18: 169,472  (eliminados: 997)
Sin educación "No sabe": 169,472  (eliminados: 0)

✅ Total eliminados: 6,704
✅ Registros finales: 169,472


In [ ]:
# 1. Duplicados:
duplicados = datos_limpio.duplicated(subset=['DIRECTORIO', 'SECUENCIA_P', 'ORDEN', 'HOGAR', 'MES']).sum()
print(f'=== DUPLICADOS ===')
print(f'Registros duplicados: {duplicados:,}')

# 2. valores atípicos en ingreso:
print(f'\n=== PERCENTILES DE INGLABO ===')
for p in [0.01, 0.05, 0.25, 0.50, 0.75, 0.95, 0.99]:
    valor = datos_limpio['INGLABO'].quantile(p)
    print(f'  Percentil {int(p*100):2d}: ${valor:,.0f}')

print(f'\n  Promedio:  ${datos_limpio["INGLABO"].mean():,.0f}')
print(f'  Mediana:   ${datos_limpio["INGLABO"].median():,.0f}')
print(f'  Máximo:    ${datos_limpio["INGLABO"].max():,.0f}')

# 3. Valores raros en otras columnas
print(f'\n=== HORAS TRABAJADAS (P6800) ===')
print(f'  Mínimo: {datos_limpio["P6800"].min()}')
print(f'  Máximo: {datos_limpio["P6800"].max()}')
print(f'  Nulos:  {datos_limpio["P6800"].isna().sum():,}')
print(f'  Ceros:  {(datos_limpio["P6800"] == 0).sum():,}')

print(f'\n=== ANTIGÜEDAD P6426 ===')
print(f'  Mínimo: {datos_limpio["P6426"].min()}')
print(f'  Máximo: {datos_limpio["P6426"].max()}')
print(f'  Nulos:  {datos_limpio["P6426"].isna().sum():,}')

=== DUPLICADOS ===
Registros duplicados: 0

=== PERCENTILES DE INGLABO ===
  Percentil  1: $80,000
  Percentil  5: $240,000
  Percentil 25: $800,000
  Percentil 50: $1,423,500
  Percentil 75: $1,850,000
  Percentil 95: $4,800,000
  Percentil 99: $9,200,000

  Promedio:  $1,746,413
  Mediana:   $1,423,500
  Máximo:    $103,000,000

=== HORAS TRABAJADAS (P6800) ===
  Mínimo: 1
  Máximo: 130
  Nulos:  0
  Ceros:  0

=== ANTIGÜEDAD P6426 ===
  Mínimo: 0
  Máximo: 850
  Nulos:  0


In [ ]:
print('=== DISTRIBUCIÓN DE HORAS SEMANALES (P6800) ===')
print(f'  Más de 80 horas: {(datos_limpio["P6800"] > 80).sum():,}')
print(f'  Más de 90 horas: {(datos_limpio["P6800"] > 90).sum():,}')
print(f'  Más de 100 horas: {(datos_limpio["P6800"] > 100).sum():,}')

print(f'\n=== ANTIGÜEDAD EN MESES (P6426) ===')
print(f'  Más de 480 meses (40 años): {(datos_limpio["P6426"] > 480).sum():,}')
print(f'  Más de 600 meses (50 años): {(datos_limpio["P6426"] > 600).sum():,}')
print(f'  Máximo: {datos_limpio["P6426"].max()} meses = {datos_limpio["P6426"].max() / 12:.0f} años')

print(f'\n=== INGRESO (INGLABO) ===')
print(f'  Más de $10,000,000: {(datos_limpio["INGLABO"] > 10_000_000).sum():,}')
print(f'  Más de $20,000,000: {(datos_limpio["INGLABO"] > 20_000_000).sum():,}')
print(f'  Más de $50,000,000: {(datos_limpio["INGLABO"] > 50_000_000).sum():,}')

=== DISTRIBUCIÓN DE HORAS SEMANALES (P6800) ===
  Más de 80 horas: 1,683
  Más de 90 horas: 433
  Más de 100 horas: 161

=== ANTIGÜEDAD EN MESES (P6426) ===
  Más de 480 meses (40 años): 1,303
  Más de 600 meses (50 años): 241
  Máximo: 850 meses = 71 años

=== INGRESO (INGLABO) ===
  Más de $10,000,000: 1,253
  Más de $20,000,000: 254
  Más de $50,000,000: 35


**Revisión adicional**
Se revisaron duplicados, outliers en ingreso, horas trabajadas y antigüedad. No se encontraron duplicados y los valores extremos representan menos del 1% del total. Se decidió conservarlos porque no afectan la mediana, que es la medida que usaremos para comparar ingresos entre grupos.

### Transformación de variables

Se convierten los valores numéricos a si categóría específica para facilitar su lectura


In [ ]:
datos_limpio['P3271'] = datos_limpio['P3271'].replace({1: 'Hombre', 2: 'Mujer'})

datos_limpio['MES'] = datos_limpio['MES'].replace({
    1: 'Enero', 2: 'Febrero', 3: 'Marzo',
    4: 'Abril', 5: 'Mayo', 6: 'Junio'
})

datos_limpio['CLASE'] = datos_limpio['CLASE'].replace({1: 'Cabecera', 2: 'Resto'})

datos_limpio['P3042'] = datos_limpio['P3042'].replace({ 1: 'Ninguno', 2: 'Preescolar', 3: 'Básica primaria (1°-5°)', 4: 'Básica secundaria (6°-9°)', 5: 'Media académica', 6: 'Media técnica', 7: 'Normalista', 8: 'Técnica profesional', 9: 'Tecnológica', 10: 'Universitaria', 11: 'Especialización', 12: 'Maestría', 13: 'Doctorado' })

datos_limpio['P6430'] = datos_limpio['P6430'].replace({
    1: 'Empleado particular', 2: 'Empleado gobierno',
    3: 'Empleado doméstico', 4: 'Cuenta propia',
    5: 'Patrón o empleador', 6: 'Familiar sin pago',
    7: 'Jornalero o peón', 8: 'Otro'
})
datos_limpio['P6440'] = datos_limpio['P6440'].replace({1: 'Sí', 2: 'No'})
datos_limpio['P6450'] = datos_limpio['P6450'].replace({1: 'Verbal', 2: 'Escrito', 9:'No sabe, no informa'})
datos_limpio['P6460'] = datos_limpio['P6460'].replace({1: 'Indefinido', 2: 'Fijo', 9:'No sabe, no informa'})
datos_limpio['P6920'] = datos_limpio['P6920'].replace({1: 'Sí', 2: 'No' , 3: 'Ya es pensionado'})
datos_limpio['P6990'] = datos_limpio['P6990'].replace({1: 'Sí', 2: 'No',9:'No sabe, no informa'})

datos_limpio['DPTO'] = datos_limpio['DPTO'].replace({
    5: 'Antioquia', 8: 'Atlántico', 11: 'Bogotá', 13: 'Bolívar',
    15: 'Boyacá', 17: 'Caldas', 18: 'Caquetá', 19: 'Cauca',
    20: 'Cesar', 23: 'Córdoba', 25: 'Cundinamarca', 27: 'Chocó',
    41: 'Huila', 44: 'La Guajira', 47: 'Magdalena', 50: 'Meta',
    52: 'Nariño', 54: 'Norte de Santander', 63: 'Quindío', 66: 'Risaralda',
    68: 'Santander', 70: 'Sucre', 73: 'Tolima', 76: 'Valle del Cauca',
    81: 'Arauca', 85: 'Casanare', 86: 'Putumayo', 88: 'San Andrés',
    91: 'Amazonas', 94: 'Guainía', 95: 'Guaviare', 97: 'Vaupés', 99: 'Vichada'
})

datos_limpio['P3069'] = datos_limpio['P3069'].replace({
    1: 'Trabaja solo', 2: '2 a 3 personas', 3: '4 a 5 personas',
    4: '6 a 10 personas', 5: '11 a 19 personas', 6: '20 a 30 personas',
    7: '31 a 50 personas', 8: '51 a 100 personas',
    9: '101 a 200 personas', 10: '201 o más personas'
})

sectores = {
    range(1, 4): 'Agricultura, ganadería, caza, silvicultura y pesca',
    range(5, 10): 'Explotación de minas y canteras',
    range(10, 34): 'Industrias manufactureras',
    range(35, 36): 'Suministro de electricidad, gas, vapor y aire acondicionado',
    range(36, 40): 'Distribución de agua; evacuación y tratamiento de aguas residuales, gestión de desechos y actividades de saneamiento ambiental',
    range(41, 44): 'Construcción',
    range(45, 48): 'Comercio al por mayor y al por menor; reparación de vehículos automotores y motocicletas',
    range(49, 54): 'Transporte y almacenamiento',
    range(55, 57): 'Alojamiento y servicios de comida',
    range(58, 64): 'Información y comunicaciones',
    range(64, 67): 'Actividades financieras y de seguros',
    range(68, 69): 'Actividades inmobiliarias',
    range(69, 76): 'Actividades profesionales, científicas y técnicas',
    range(77, 83): 'Actividades de servicios administrativos y de apoyo',
    range(84, 85): 'Administración pública y defensa; planes de seguridad social de afiliación obligatoria',
    range(85, 86): 'Educación',
    range(86, 89): 'Actividades de atención de la salud humana y de asistencia social',
    range(90, 94): 'Actividades artísticas, de entretenimiento y recreación ',
    range(94, 97): 'Otras actividades de servicios',
    range(97, 99): 'Actividades de los hogares en calidad de empleadores; actividades no diferenciadas de los hogares individuales como productores de bienes y servicios para uso propio',
    range(99, 100): 'Actividades de organizaciones y entidades extraterritoriales',
}

def clasificar_sector(codigo):
    if pd.isna(codigo):
        return 'No especificado'
    codigo = int(codigo)
    for rango, nombre in sectores.items():
        if codigo in rango:
            return nombre
    return 'No clasificado'

datos_limpio['RAMA2D_R4'] = datos_limpio['RAMA2D_R4'].apply(clasificar_sector)

# Renombrar columnas
datos_limpio = datos_limpio.rename(columns={
    'P3271': 'SEXO',
    'P6040': 'EDAD',
    'P3042': 'NIVEL_EDUCATIVO',
    'P6430': 'POSICION',
    'P6440': 'TIENE_CONTRATO',
    'P6450': 'TIPO_CONTRATO',
    'P6460': 'DURACION_CONTRATO',
    'P6800': 'HORAS_SEMANA',
    'P6920': 'COTIZA_PENSION',
    'P6990': 'TIENE_ARL',
    'P6426': 'ANTIGUEDAD_MESES',
    'P3069':'TAMAÑO_EMPRESA',
    'DPTO': 'DEPARTAMENTO',
    'RAMA2D_R4': 'SECTOR_ECONOMICO',

})



In [ ]:
datos_limpio.head(3)

,DIRECTORIO,SECUENCIA_P,ORDEN,HOGAR,MES,CLASE,DEPARTAMENTO,FEX_C18,INGLABO,HORAS_SEMANA,...,TIENE_CONTRATO,TIPO_CONTRATO,DURACION_CONTRATO,COTIZA_PENSION,TIENE_ARL,ANTIGUEDAD_MESES,TAMAÑO_EMPRESA,SEXO,EDAD,NIVEL_EDUCATIVO
0,8086872,1,1,1,Enero,Resto,Bolívar,2184.752283,1040000.0,24,...,Sí,Verbal,NaN,No,No,10,Trabaja solo,Mujer,54,Básica primaria (1°-5°)
1,8086874,1,2,1,Enero,Resto,Bolívar,1208.490211,700000.0,60,...,No,NaN,NaN,No,No,480,Trabaja solo,Hombre,60,Básica secundaria (6°-9°)
2,8086874,1,3,1,Enero,Resto,Bolívar,1208.490211,1450000.0,50,...,Sí,Verbal,NaN,No,No,12,2 a 3 personas,Hombre,59,Básica secundaria (6°-9°)


In [ ]:
# Columnas categóricas
categoricas = ['SEXO', 'MES', 'CLASE', 'NIVEL_EDUCATIVO', 'POSICION',
               'TIENE_CONTRATO', 'TIPO_CONTRATO', 'DURACION_CONTRATO','SECTOR_ECONOMICO',
               'COTIZA_PENSION', 'TIENE_ARL', 'TAMAÑO_EMPRESA','DEPARTAMENTO']

for col in categoricas:
    print(f'=== {col} ===')
    print(datos_limpio[col].value_counts())
    print()

# Columnas numéricas
print('=== COLUMNAS NUMÉRICAS ===')
numericas = ['EDAD', 'INGLABO', 'HORAS_SEMANA', 'ANTIGUEDAD_MESES']
for col in numericas:
    print(f'{col}: min={datos_limpio[col].min()}, max={datos_limpio[col].max()}, nulos={datos_limpio[col].isna().sum()}')

=== SEXO ===
SEXO
Hombre    94835
Mujer     74637
Name: count, dtype: int64

=== MES ===
MES
Mayo       28929
Junio      28717
Abril      28462
Marzo      28290
Febrero    27970
Enero      27104
Name: count, dtype: int64

=== CLASE ===
CLASE
Cabecera    149609
Resto        19863
Name: count, dtype: int64

=== NIVEL_EDUCATIVO ===
NIVEL_EDUCATIVO
Media académica              54044
Básica primaria (1°-5°)      29039
Universitaria                27429
Básica secundaria (6°-9°)    17645
Técnica profesional          15917
Tecnológica                   7454
Especialización               5909
Ninguno                       4279
Media técnica                 4182
Maestría                      2884
Doctorado                      421
Normalista                     256
Preescolar                      13
Name: count, dtype: int64

=== POSICION ===
POSICION
Cuenta propia          73717
Empleado particular    73190
Empleado gobierno       8980
Empleado doméstico      5910
Jornalero o peón        3940


### Relacionar departamentos con sus coordenadas Latitud y Longitud

In [ ]:
# Agregar coordenadas al dataset principal
coordenadas_df = pd.read_excel(
    '/content/drive/MyDrive/Factores que determinan el ingreso laboral en Colombia/Coordenadas_Departamentos.xlsx',
    skiprows=9,
    nrows=33
)

# Normalizar nombres para el cruce
coordenadas_df['NOMBRE_CRUCE'] = coordenadas_df['Nombre'].str.strip().str.upper()
equivalencias = {
    'BOGOTÁ, D.C.': 'BOGOTÁ',
    'ARCHIPIÉLAGO DE SAN ANDRÉS, PROVIDENCIA Y SANTA CATALINA': 'SAN ANDRÉS',
}
coordenadas_df['NOMBRE_CRUCE'] = coordenadas_df['NOMBRE_CRUCE'].replace(equivalencias)

coord_merge = coordenadas_df[['NOMBRE_CRUCE', 'LATITUD', 'LONGITUD']].copy()
datos_limpio['NOMBRE_CRUCE'] = datos_limpio['DEPARTAMENTO'].str.upper()
datos_limpio = datos_limpio.merge(coord_merge, on='NOMBRE_CRUCE', how='left')
datos_limpio = datos_limpio.drop(columns=['NOMBRE_CRUCE'])

# Verificar
print(f'Nulos en LATITUD: {datos_limpio["LATITUD"].isna().sum()}')
print(f'Nulos en LONGITUD: {datos_limpio["LONGITUD"].isna().sum()}')

Nulos en LATITUD: 0
Nulos en LONGITUD: 0


In [ ]:
datos_limpio.head()

,DIRECTORIO,SECUENCIA_P,ORDEN,HOGAR,MES,CLASE,DEPARTAMENTO,FEX_C18,INGLABO,HORAS_SEMANA,...,DURACION_CONTRATO,COTIZA_PENSION,TIENE_ARL,ANTIGUEDAD_MESES,TAMAÑO_EMPRESA,SEXO,EDAD,NIVEL_EDUCATIVO,LATITUD,LONGITUD
0,8086872,1,1,1,Enero,Resto,Bolívar,2184.752283,1040000.0,24,...,NaN,No,No,10,Trabaja solo,Mujer,54,Básica primaria (1°-5°),8.079797,-74.235148
1,8086874,1,2,1,Enero,Resto,Bolívar,1208.490211,700000.0,60,...,NaN,No,No,480,Trabaja solo,Hombre,60,Básica secundaria (6°-9°),8.079797,-74.235148
2,8086874,1,3,1,Enero,Resto,Bolívar,1208.490211,1450000.0,50,...,NaN,No,No,12,2 a 3 personas,Hombre,59,Básica secundaria (6°-9°),8.079797,-74.235148
3,8086876,1,2,1,Enero,Resto,Bolívar,2420.064922,350000.0,40,...,NaN,No,No,216,Trabaja solo,Hombre,45,Básica secundaria (6°-9°),8.079797,-74.235148
4,8086876,1,3,1,Enero,Resto,Bolívar,2420.064922,1350000.0,50,...,NaN,No,No,144,2 a 3 personas,Hombre,44,Media académica,8.079797,-74.235148


## Guardar dataset limpio

In [ ]:
ruta_salida = '/content/drive/MyDrive/Factores que determinan el ingreso laboral en Colombia/geih_limpio.csv'
datos_limpio.to_csv(ruta_salida, index=False)

size_mb = os.path.getsize(ruta_salida) / (1024 * 1024)
print(f'✅ Dataset guardado en: {ruta_salida}')
print(f'   Filas: {len(datos_limpio):,}')
print(f'   Columnas: {datos_limpio.shape[1]}')
print(f'   Tamaño: {size_mb:.1f} MB')

✅ Dataset guardado en: /content/drive/MyDrive/Factores que determinan el ingreso laboral en Colombia/geih_limpio.csv
   Filas: 169,472
   Columnas: 24
   Tamaño: 36.9 MB
